src/librawspeed/decoders/NakedDecoder.cpp

```diff
@@ -44,50 +44,55 @@ NakedDecoder::NakedDecoder(FileMap* file, Camera* c) :
 
 NakedDecoder::~NakedDecoder() = default;
 
-RawImage NakedDecoder::decodeRawInternal() {
-  uint32 width=0, height=0, filesize=0, bits=0, offset=0;
-  if(cam->hints.find("full_width") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_width"))->second;
-    width = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find width");
-
-  if(cam->hints.find("full_height") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_height"))->second;
-    height = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find height");
-
-  if(cam->hints.find("filesize") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("filesize"))->second;
-    filesize = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find filesize");
-
-  if(cam->hints.find("offset") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("offset"))->second;
-    offset = (uint32) atoi(tmp.c_str());
-  }
+static const map<string, BitOrder> order2enum = {
+    {"plain", BitOrder_Plain},
+    {"jpeg", BitOrder_Jpeg},
+    {"jpeg16", BitOrder_Jpeg16},
+    {"jpeg32", BitOrder_Jpeg32},
+};
+
+void NakedDecoder::parseHints() {
```

Comment:
May I ask why you decided to move the code into a separate function here? I would have done exactly the opposite if I found this (new) code and wanted to improve it. My reasoning would be that it is called only from one place and that is totally obvious when it written in this one place. Now one has to grep the whole file to know what might be affected by changing something. You also widened the scope of the 6 former local variables to be now class variables. Again: if someone wants to understand what they are doing he now has to inspect all of the member functions instead of only the one where they are actually used.